src/perform_step/gpu_em_perform_step.jl

```diff
@@ -0,0 +1,49 @@
+function em_kernel(probs, _us, _ts, dt,
+                   saveat, ::Val{save_everystep}) where {save_everystep}
+    i = (blockIdx().x - 1) * blockDim().x + threadIdx().x
+    i > length(probs) && return
+
+    # get the actual problem for this thread
+    prob = @inbounds probs[i]
+
+    # get the input/output arrays for this thread
+    ts = @inbounds view(_ts, :, i)
+    us = @inbounds view(_us, :, i)
+
+    f = prob.f
+    g = prob.g
+    u0 = prob.u0
+    tspan = prob.tspan
+    p = prob.p
+
+    @inbounds ts[1] = prob.tspan[1]
+    @inbounds us[1] = prob.u0
+
+    sqdt = sqrt(dt)
+    u = copy(u0)
+    t = copy(tspan[1])
+    n = length(ts)
+
+    for j in 2:n
+        uprev = u
+
+        u = uprev + f(uprev, p, t) * dt +
+            sqdt * g(uprev, p, t) .* randn(typeof(u0))
```

Comment:
This `.*` needs to be expanded to support non-diagonal, but that can happen in a follow up